app/controllers/application_controller.rb

```diff
@@ -1,10 +1,84 @@
 require './config/environment'
-
+require 'pry'
 class ApplicationController < Sinatra::Base
 
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "password_security"
+  end
+
+  get '/' do 
+    erb :index
+  end
+
+  get '/signup' do
+    # binding.pry
+    if !logged_in 
+      erb :signup
+    else
+     redirect '/tweets'
+    end
+  end
+
+  post '/signup' do
+    if valid_new_user(params)
+      @user = User.new(username: params[:username],email: params[:email],password_digest: BCrypt::Password.create(params[:password]))
```

Comment:
I think if you pass `params[:password]` to the `password` attribute, it should automatically convert it to the password digest.